src/coreclr/tools/Common/JitInterface/CorInfoInstructionSet.cs

```diff
@@ -240,54 +239,85 @@ public enum InstructionSet_X86
         X86Serialize_X64 = 40,
     }
 
-    public struct InstructionSetFlags : IEnumerable<InstructionSet>
+    public unsafe struct InstructionSetFlags : IEnumerable<InstructionSet>
     {
-        private ulong _flags;
-
+        private const int _flagsArrSize = 1;
+        private fixed ulong _flags[_flagsArrSize];
         public IEnumerable<InstructionSet_ARM64> ARM64Flags => this.Select((x) => (InstructionSet_ARM64)x);
 
         public IEnumerable<InstructionSet_X64> X64Flags => this.Select((x) => (InstructionSet_X64)x);
 
         public IEnumerable<InstructionSet_X86> X86Flags => this.Select((x) => (InstructionSet_X86)x);
 
+        public InstructionSetFlags() { }
+
         public void AddInstructionSet(InstructionSet instructionSet)
         {
-            _flags = _flags | (((ulong)1) << (int)instructionSet);
+            int arrayIdx = (int)instructionSet / 64;
```

Comment:
It is better to use uint in this pattern. Signed division is less efficient.

Reply:
Ah. Good  catch- it's a power of 2 division. Changed